<?xml version="1.0"?>
<launch> 
  <param name="use_sim_time" type="bool" value="false"/>
  
  <!-- load workspace constraints -->
  <rosparam file="$(find hiqp_demos)/config/yumi_preload.yaml" command="load"/>
  
  <rosparam file="$(find demo_teleop)/config/yumi_left_fulldof.yaml" command="load" ns="/demo_teleop" />

  <include file="$(find hiqp_demos)/launch/yumi_online.launch">
    <arg name="rviz" value="false"/>
  </include>

  <node name="demo_teleop" pkg="demo_teleop" type="demo_teleop_node" respawn="false" output="screen" >
     <param name="frame_left" type="string" value="gripper_l_base"/>
     <param name="frame_right" type="string" value="gripper_r_base"/>
     
     <param name="run_online" value="false"/>
     <rosparam param="jnts_teleop_sensing">[ -0.4, -1.88, 0.61, 0.19, -0.84, 0.15, -2.12, 0.52, -1.94, -0.65, 0.28, 1.58, 0.18, -1.52]</rosparam>
     <rosparam param="jnts_teleop_init">[-0.84, -2.13, 1.47, 0.89, 1.15, 0.42, -3.0, 0.84, -2.13, -1.47, 0.89, -1.15, 0.42, 0.0]</rosparam>
  </node>
  
  <node pkg="leap_motion" type="leap_hands" name="leap_hands" output="screen">
    <param name="frame_name" value="/leap_optical_frame" />
    <param name="publish_frame_name" value="yumi_body" />
    <param name="left_frame_name" value="teleop_left_frame" />
    <param name="right_frame_name" value="teleop_right_frame" />
  </node>

  <node pkg="tf" type="static_transform_publisher" name="leap_link"
        args="0.4 0.0 -0.02 0.0 0.0 0.0 /yumi_body /leap_optical_frame 50" />

  <!--include file="$(find grasp_planner)/launch/test_yumi.launch">
  </include-->
  
  <!--include file="$(find hiqp_ros)/launch/yumi_simulation.launch">
    <arg name="start_rviz" value="false" />
  </include-->
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find demo_teleop)/launch/yumi.rviz"/>


</launch>
